index.js

```diff
@@ -71,7 +71,15 @@ function redisStore(args) {
       }
 
       if (opts.parse) {
-        result = JSON.parse(result);
+
+        try {
+          // allow undefined only if allowed by isCacheableValue
+          if(! ( (result === undefined || result === "undefined") && typeof args.isCacheableValue === 'function' && args.isCacheableValue(result))) {
```

Comment:
The function handleRespone is used for get and set, running our test suite showed both undefined and "undefined" need to be handled. I'll try to pinpoint it if needed.

Every store (as a plugin) can provide it's isCacheableValue but since it is a plugin to cache-manager, the cache manager should mandate if stuff that is not cacheable should be stored ( which it shouldn't as this is the whole point). If a value should not be stored, we should receive an error in callback (not swallow it) For example if we do a set(key, undefined) we (now with this PR) get an error since undefined is not allowed by default. I don't think it is good to allow a set to go without error and then get null (for the same key) as there is no indication of success nor failure and user can think the save succeeded, when it fact it did not.
The other edge case here is if the user allows undefined by isCachebaleValue and then this gets honored and value stored to db as "undefined" and retrieved as such.
I this PR we have test cases for both.